github-actions/feature-request/src/get-input.ts

```diff
@@ -0,0 +1,28 @@
+import * as core from '@actions/core';
+
+interface InputProvider {
+  getInput(name: string): string;
+}
+
+// Gets a specific value from the YAML configuration.
+// The value could be either a number or a string.
+export const getInputValue = <T extends number | string | boolean>(
+  name: string,
+  _core: InputProvider = core,
+): T => {
+  const result = _core.getInput(name);
+  if (!result) {
+    throw new Error(`No value for ${name} specified in the configuration.`);
+  }
+  if (/^(true|false)$/.test(result)) {
+    return (result === 'true' ? true : false) as T;
+  }
+  if (!/^-?\d+(\.?\d*)$/.test(result)) {
+    return result as T;
+  }
+  const num = parseFloat(result);
+  if (isNaN(num)) {
+    throw new Error(`Can't parse ${name} as a numeric value.`);
+  }
+  return num as T;
```

Comment:
```suggestion
  if (!isNaN(result)) {
    return parseFloat(result) as T;
  }
  return result as T;
```

Reply:
Interesting! Did not know `isNaN` casts to number if we pass a string.

Reply:
TypeScript is unhappy though because `isNaN` expects a number.